base/essentials.jl

```diff
@@ -204,7 +204,7 @@ tail(::Tuple{}) = throw(ArgumentError("Cannot call tail on an empty tuple."))
 tuple_type_head(T::Type) = (@_pure_meta; fieldtype(T::Type{<:Tuple}, 1))
 
 function tuple_type_tail(T::Type)
-    @_pure_meta
+    @_pure_meta # TODO: this method is wrong (and not @pure)
```

Comment:
not pure because it `throw`s? or because it's defined on `T::Type`?

Reply:
Yeah, it would be nice if you could expand on how this is wrong and impure in the comment. That would make it easier for others to revisit and attempt to fix later.